### ISSUE TITLE
Decimals cause an error

### ISSUE THREAD

**Issue by cjhenck** [segmentio/analytics-python#63]
Decimals are not serialized by the built in json package, but the code in analytics-python/analytics/utils.py reads:

```python
def clean(item):
    if isinstance(item, (six.string_types, bool, numbers.Number, datetime,
                         type(None))):
        return item
    elif isinstance(item, (set, list, tuple)):
        return _clean_list(item)
    elif isinstance(item, dict):
        return _clean_dict(item)
    else:
        return _coerce_unicode(item)
```

Decimal is a numeric type so it is returned, but it should be cleaned somehow first.  Ideally this would be a custom JSONEncoder subclass (so that the string value of the JSON will be sent), but a float() conversion would help.

**Comment by adamchainz** [comment]
`float` would be bad since it loses precision

**Comment by cjhenck** [comment]
`float` is definitely not the ideal solution, but as it is right now Decimals kill the serialization completely.  The proper thing would be to add a JSONEncoder subclass that handles Decimals as I mentioned.  From our end floats were OK for analytics purposes since segment doesn't feed into any finance systems for us (after all the messages are not guaranteed)

**Comment by jgershen** [comment]
Thanks for noticing and fixing this @cjhenck. I've included your patch in version 1.2.4. The point about subclassing JSONEncoder is noted for the future, too.
